frontend/src/api/messagingQueues/onboarding/getOnboardingStatus.ts

```diff
@@ -0,0 +1,37 @@
+import { ApiBaseInstance } from 'api';
+import { ErrorResponseHandler } from 'api/ErrorResponseHandler';
+import { AxiosError } from 'axios';
+import { SOMETHING_WENT_WRONG } from 'constants/api';
+import { ErrorResponse, SuccessResponse } from 'types/api';
+
+export interface OnboardingStatusResponse {
+	status: string;
+	data: {
+		attribute?: string;
+		error_message?: string;
+		status?: string;
+	}[];
+}
+
+const getOnboardingStatus = async (props: {
+	start: number;
+	end: number;
+}): Promise<SuccessResponse<OnboardingStatusResponse> | ErrorResponse> => {
+	try {
```

Comment:
@SagarRajput-7 : have we tested for error cases ?

Reply:
yes, the error are consumed, and we have retry polling there in case of either error or a valid response and post that the ui will show that `failed`